crypto/stack/stack.c

```diff
@@ -322,17 +327,36 @@ static int internal_find(OPENSSL_STACK *st, const void *data,
     r = ossl_bsearch(&data, st->data, st->num, sizeof(void *), st->comp,
                      ret_val_options);
 
+    if (pnum != NULL) {
+        *pnum = 0;
+        if (r != NULL) {
+            const void **p = (const void **)r;
+
+            while (p < st->data + st->num) {
```

Comment:
This implementation assumes that the flags include `OSSL_BSEARCH_FIRST_VALUE_ON_MATCH` - which they do for the only caller where `pnum != NULL`...but perhaps at least a comment is in order. Or possibly an assertion?

Reply:
Makes sense. Fixup pushed to forcibly add the flag.